### PR TITLE
Removes all wrapped errors in offline.go

### PR DESCRIPTION
**Pull request by epps**
# Description

Removes all wrapped errors in offline.go.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/offline.go [hunk 1]
```diff
@@ -79,11 +80,11 @@ func (id *ResourceID) FromFilestorePath(path string) error {
 	featureformRootPathPart := "featureform/"
 	idx := strings.Index(path, featureformRootPathPart)
 	if idx == -1 {
-		return fmt.Errorf("expected \"featureform\" root path part in path %s", path)
+		return fferr.NewInvalidArgument(fmt.Errorf("expected \"featureform\" root path part in path %s", path))
```
Comment (sdreyer) [inline provider/offline.go:83]: Think this might be internal since we set the path

File: provider/offline.go [hunk 2]
```diff
@@ -79,11 +80,11 @@ func (id *ResourceID) FromFilestorePath(path string) error {
...
 	}
 	resourceParts := strings.Split(path[idx+len(featureformRootPathPart):], "/")
 	if len(resourceParts) < 3 {
-		return fmt.Errorf("expected path %s to contain OfflineResourceType/Name/Variant", strings.Join(resourceParts, "/"))
+		return fferr.NewInvalidArgument(fmt.Errorf("expected path %s to contain OfflineResourceType/Name/Variant", strings.Join(resourceParts, "/")))
```
Comment (sdreyer) [inline provider/offline.go:87]: think this is internal

File: provider/offline.go [hunk 3]
```diff
@@ -99,7 +100,7 @@ func (id *ResourceID) FromFilestorePath(path string) error {
 	case "Materialization":
 		id.Type = OfflineResourceType(6)
 	default:
-		return fmt.Errorf("unrecognized OfflineResourceType: %s", resourceParts[0])
+		return fferr.NewInvalidArgument(fmt.Errorf("unrecognized OfflineResourceType: %s", resourceParts[0]))
```
Comment (sdreyer) [inline provider/offline.go:103]: think this is internal

File: provider/offline.go [hunk 4]
```diff
@@ -372,7 +372,7 @@ func (rec *ResourceRecord) SetEntity(entity interface{}) error {
 	case string:
 		rec.Entity = typedEntity
 	default:
-		return fmt.Errorf("entity must be a string; received %T", entity)
+		return fferr.NewInvalidArgument(fmt.Errorf("entity must be a string; received %T", entity))
```
Comment (sdreyer) [inline provider/offline.go:375]: Internal or type error?

File: provider/offline.go [hunk 5]
```diff
@@ -590,39 +590,39 @@ func (store *memoryOfflineStore) AsOfflineStore() (OfflineStore, error) {
 }
 
 func (store *memoryOfflineStore) RegisterResourceFromSourceTable(id ResourceID, schema ResourceSchema) (OfflineTable, error) {
-	return nil, fmt.Errorf("Snowflake RegisterResourceFromSourceTable not implemented")
+	return nil, fferr.NewInternalError(fmt.Errorf("not implemented"))
 }
 
 func (store *memoryOfflineStore) RegisterPrimaryFromSourceTable(id ResourceID, sourceName string) (PrimaryTable, error) {
-	return nil, fmt.Errorf("Snowflake RegisterPrimaryFromSourceTable not implemented")
```
Comment (sdreyer) [inline provider/offline.go:597]: lol

File: provider/offline.go (hunk 4 above)
Comment (epps) [reply provider/offline.go:375]: I actually think it's fine to use invalid arg given the user is the one who's responsible for this error.

Comment (sdreyer) [reply provider/offline.go:375]: thats fine

**Review by sdreyer — APPROVED** [review]

**Pull request merged**